### PR TITLE
backup: support gcp backup with br

### PR DESCRIPTION
**Pull request by shuijing198799**
### What problem does this PR solve? 
support gcp backup with br
### What is changed and how does it work?
check bucket and return gcp args for support br backup to gcs
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Go code change

```release-note
support backup and restore with GCS using BR
add field `prefix` in field `gcsStorageProvider` for define the prefix in gcs bucket
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -869,6 +869,8 @@ type GcsStorageProvider struct {
 	// SecretName is the name of secret which stores the
 	// gcs service account credentials JSON .
 	SecretName string `json:"secretName"`
+	// Prefix for the keys.
+	Prefix string `json:"prefix,omitempty"`
```
Comment (Yisaer) [inline pkg/apis/pingcap/v1alpha1/types.go:873]: Plz provide some note comments to explain the `Prefix` here.

**Comment by DanielZhangQD** [comment]
Suggest changes for release notes:
```
Support BR backup and restore with GCS
```

File: cmd/backup-manager/app/util/remote.go [hunk 2]
```diff
@@ -159,3 +229,23 @@ func checkS3Config(s3 *v1alpha1.S3StorageProvider, fakeRegion bool) *s3Query {
 
 	return &sqs
 }
+
+// checkS3Config constructs s3Query parameters
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/util/remote.go:233]: Change the comments.

File: cmd/backup-manager/app/util/remote.go [hunk 3]
```diff
@@ -159,3 +229,23 @@ func checkS3Config(s3 *v1alpha1.S3StorageProvider, fakeRegion bool) *s3Query {
...
+
+	gqs.bucket = gcs.Bucket
+	gqs.location = gcs.Location
+	gqs.path = gcs.Path
+	gqs.projectId = gcs.ProjectId
+	gqs.storageClass = gcs.StorageClass
+	gqs.objectAcl = gcs.ObjectAcl
+	gqs.bucketAcl = gcs.BucketAcl
+	gqs.secretName = gcs.SecretName
+	gqs.prefix = gcs.Prefix
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/util/remote.go:245]: The `Location`, `Path`, `ProjectID`, `BucketAcl` and `SecretName` are not used?

Comment (shuijing198799) [reply cmd/backup-manager/app/util/remote.go:245]: temporary not used

File: cmd/backup-manager/app/util/remote.go (hunk 2 above)
Comment (shuijing198799) [reply cmd/backup-manager/app/util/remote.go]: done

File: pkg/apis/pingcap/v1alpha1/types.go (hunk 1 above)
Comment (shuijing198799) [reply pkg/apis/pingcap/v1alpha1/types.go:873]: done

**Comment by shuijing198799** [comment]
> Suggest changes for release notes:
> 
> ```
> Support backup and restore with GCS using BR
> ```

done

File: cmd/backup-manager/app/util/remote.go [hunk 4]
```diff
@@ -230,7 +230,7 @@ func checkS3Config(s3 *v1alpha1.S3StorageProvider, fakeRegion bool) *s3Query {
 	return &sqs
 }
 
-// checkS3Config constructs s3Query parameters
+// checkGcsConfig constructs s3Query parameters
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/util/remote.go:233]:
```suggestion
// checkGcsConfig constructs gcsQuery parameters
```

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 5]
```diff
@@ -840,7 +840,7 @@ type S3StorageProvider struct {
 	// SecretName is the name of secret which stores
 	// S3 compliant storage access key and secret key.
 	SecretName string `json:"secretName,omitempty"`
-	// Prefix for the keys.
+	// Folder of the bucket.
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go:843]:
```suggestion
	// Prefix of the data path.
```

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 6]
```diff
@@ -867,9 +867,9 @@ type GcsStorageProvider struct {
 	// BucketAcl represents the access control list for new buckets
 	BucketAcl string `json:"bucketAcl,omitempty"`
 	// SecretName is the name of secret which stores the
-	// gcs service account credentials JSON .
+	// gcs service account credentials JSON.
 	SecretName string `json:"secretName"`
-	// Prefix for the keys.
+	// Folder of the bucket.
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go:872]:
```suggestion
	// Prefix of the data path.
```

File: pkg/apis/pingcap/v1alpha1/types.go (hunk 5 above)
Comment (shuijing198799) [reply pkg/apis/pingcap/v1alpha1/types.go]: done

File: cmd/backup-manager/app/util/remote.go (hunk 4 above)
Comment (shuijing198799) [reply cmd/backup-manager/app/util/remote.go]: done

File: pkg/apis/pingcap/v1alpha1/types.go (hunk 6 above)
Comment (shuijing198799) [reply pkg/apis/pingcap/v1alpha1/types.go]: done

**Pull request merged**